py-scripts/lf_interop_throughput.py

```diff
@@ -1941,6 +2199,164 @@ def generate_report(self, iterations_before_test_stopped_by_user, incremental_ca
         report.build_footer()
         report.write_html()
         report.write_pdf(_orientation="Landscape")
+    # Creates a separate DataFrame for each group of devices.
+
+    def generate_dataframe(self, groupdevlist, typeofdevice, devusername, devssid, devmac, devchannel, devmode, devdirection, devofdownload, devobsdownload,
+                           devoffupload, devobsupload, devRSSI, devExpected, devlinkspeed, devpacketsize, devstatus, upload_drop, download_drop):
+        """
+        Creates a separate DataFrame for each group of devices.
+
+        Returns:
+            DataFrame: A DataFrame for each device group.
+            Returns None if neither device in a group is configured.
+        """
+
+        device_type = []
+        username = []
+        ssid = []
+        mac = []
+        channel = []
+        mode = []
+        direction = []
+        offdownload = []
+        obsdownload = []
+        offupload = []
+        obsupload = []
+        RSSI = []
```

Comment:
Variables should be lowercase. Per PEP style guidelines, only constants should be capitalized.